cmd/resourcegen/embed/resources/$cloud_provider$/$filename$.go.tmpl

```diff
@@ -71,66 +79,26 @@ var {{ .ResourceName }}UsageSchema = []*schema.UsageItem{
 
 // PopulateUsage parses the u schema.UsageData into the {{ .ResourceName }}.
 // It uses the `infracost_usage` struct tags to populate data into the {{ .ResourceName }}.
-func ({{ .ResourceLetter }} *{{ .ResourceName }}) PopulateUsage(u *schema.UsageData) {
-	resources.PopulateArgsWithUsage({{ .ResourceLetter }}, u)
+func (r *{{ .ResourceName }}) PopulateUsage(u *schema.UsageData) {
+	resources.PopulateArgsWithUsage(r, u)
 }
 
-// BuildResource builds a schema.Resource from a valid {{ .ResourceName }}.
-// This method is called after the resource is initialised by an iac provider.
+// BuildResource builds a schema.Resource from a valid {{ .ResourceName }} struct.
+// This method is called after the resource is initialised by an IaC provider.
 // See providers folder for more information.
-func ({{ .ResourceLetter }}  *{{.ResourceName}}) BuildResource() *schema.Resource {
+func (r *{{.ResourceName}}) BuildResource() *schema.Resource {
```

Comment:
Think removing `ResourceLetter` is probably a good idea as it simplifies things. But just want to make sure we're happy that all resources will have a `r` as receiver name. i.e:


```go

func (r AResource) BuildResource() {
}
```

It's kinda convention to have this start with the same letter as the struct (in the example case it would be `a`. However, I don't think it matters too much and cleans up a lot of code generation code.

Reply:
I'm still new to Go naming conventions. But the idea behind is to make the code look similar between resources as much as possible. The goal is to create a scaffold where only mapping logic and cost components differ.
And as we already have `d schema.ResourceData` inconsistency, I thought we could do the same for the struct name.